src/index.ts

```diff
@@ -1,105 +1,99 @@
 /*
  This file contains functions to sign message for executeRelayCall.
 */
-import Account from 'eth-lib/lib/account';
-import { bufferToHex, keccak256 } from 'ethereumjs-util';
-import utils from 'web3-utils';
-
-import { Message } from './interfaces';
+import {
+  type Address,
+  hexToBytes,
+  isAddress,
+  isHex,
+  keccak256,
+  recoverAddress,
+  serializeSignature,
+  stringToHex,
+} from 'viem';
+import { sign } from 'viem/accounts';
 
+import type { Message } from './interfaces';
 export class EIP191Signer {
-  hashEthereumSignedMessage(message: string) {
-    const messageHex = utils.isHexStrict(message)
-      ? message
-      : utils.utf8ToHex(message);
-    const messageBytes = utils.hexToBytes(messageHex);
-    const messageBuffer = Buffer.from(messageBytes);
-    const preamble =
-      '\x19' + '\x45' + 'thereum Signed Message:\n' + messageBytes.length;
-    const preambleBuffer = Buffer.from(preamble);
-    const ethMessage = Buffer.concat([preambleBuffer, messageBuffer]);
-    return bufferToHex(keccak256(ethMessage));
+  hashEthereumSignedMessage(message) {
+    const messageHex = isHex(message) ? message : stringToHex(message);
+    const messageBytes = hexToBytes(messageHex);
+    const encoder = new TextEncoder();
+    const preambleBytes = encoder.encode(
+      `\x19\x45thereum Signed Message:\n${messageBytes.length}`,
+    );
+    const ethMessage = new Uint8Array(
+      preambleBytes.length + messageBytes.length,
+    );
+    ethMessage.set(preambleBytes, 0); // Add the byte array starting at index 0
+    ethMessage.set(messageBytes, preambleBytes.length); // Add the string bytes after
+    return keccak256(ethMessage, 'hex');
   }
-
-  hashDataWithIntendedValidator(validatorAddress: string, data: string) {
+  hashDataWithIntendedValidator(validatorAddress, data) {
     // validator address
-    if (!utils.isAddress(validatorAddress))
+    if (!isAddress(validatorAddress)) {
       throw new Error('Validator needs to be a valid address');
-
-    const validatorBuffer = Buffer.from(utils.hexToBytes(validatorAddress));
+    }
+    const validatorBuffer = hexToBytes(validatorAddress);
```

Comment:
We could remove Buffer from the name variable

Reply:
Ok changed the variables to nameBytes instead.